tf2_web_republisher/src/tf2_web_republisher.cpp

```diff
@@ -0,0 +1,138 @@
+  tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
+}
+
+rclcpp_action::CancelResponse TFRepublisher::handle_cancel(
+    const std::shared_ptr<rclcpp_action::ServerGoalHandle<tf2_web_republisher_msgs::action::TFSubscription>> /*gh*/)
+{
+  RCLCPP_DEBUG(rclcpp::get_logger("tf2_web_republisher"), "GoalHandle canceled");
+  return rclcpp_action::CancelResponse::ACCEPT;
+}
+
+std::string TFRepublisher::cleanTfFrame(const std::string& frame_id) const
+{
+  if (frame_id[0] == '/')
+  {
+    return frame_id.substr(1);
+  }
+  return frame_id;
+}
+
+std::optional<geometry_msgs::msg::TransformStamped> TFRepublisher::threadSafeLookup(const std::string& target_frame,
+                                                                                    const std::string& source_frame)
+{
+  std::scoped_lock<std::mutex> lock(tf_buffer_mutex_);
+  std::optional<geometry_msgs::msg::TransformStamped> out;
+  try
+  {
+    out = tf_buffer_->lookupTransform(cleanTfFrame(target_frame), cleanTfFrame(source_frame), tf2::TimePointZero);
+  }
+  catch (const tf2::TransformException& ex)
+  {
+    RCLCPP_INFO(this->get_logger(), "Could not transform %s to %s: %s", target_frame.c_str(), source_frame.c_str(),
+                ex.what());
+  }
+
+  return out;
+}
+
+rclcpp_action::GoalResponse
+TFRepublisher::handle_goal(const rclcpp_action::GoalUUID& /*uuid*/,
+                           const std::shared_ptr<const tf2_web_republisher_msgs::action::TFSubscription::Goal>& /*goal*/)
+{
+  RCLCPP_DEBUG(rclcpp::get_logger("tf2_web_republisher"), "GoalHandle request received");
+  return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
+}
+
+void TFRepublisher::handle_accepted(
+    const std::shared_ptr<rclcpp_action::ServerGoalHandle<tf2_web_republisher_msgs::action::TFSubscription>>& goal_handle)
+{
+  using namespace std::placeholders;
+  // this needs to return quickly to avoid blocking the executor, so spin up a new thread
+  std::thread{ [this](auto&& PH1) { execute(std::forward<decltype(PH1)>(PH1)); }, goal_handle }.detach();
+}
+
+void TFRepublisher::execute(
+    const std::shared_ptr<rclcpp_action::ServerGoalHandle<tf2_web_republisher_msgs::action::TFSubscription>>& goal_handle)
+{
+  auto feedback = std::make_shared<tf2_web_republisher_msgs::action::TFSubscription::Feedback>();
+
+  std::string target_frame = goal_handle->get_goal()->target_frame;
+  feedback->transforms.reserve(goal_handle->get_goal()->source_frames.size());
+  while (!goal_handle->is_canceling())
+  {
+    feedback->transforms.resize(0);
+    for (auto& source_frame : goal_handle->get_goal()->source_frames)
+    {
+      if (auto transform = threadSafeLookup(target_frame, source_frame))
+      {
+        feedback->transforms.push_back(transform.value());
+      }
+    }
+    // publish feedback
+    goal_handle->publish_feedback(feedback);
+    rclcpp::sleep_for(std::chrono::nanoseconds((size_t)(1E9 / goal_handle->get_goal()->rate)));
```

Comment:
I suggest using `static_cast<>` instead of C-style cast.
```suggestion
    rclcpp::sleep_for(std::chrono::nanoseconds(static_cast<size_t>(1E9 / goal_handle->get_goal()->rate)));
```
This is a nitpick though.